src/settings/project/mod.rs

```diff
@@ -355,5 +358,40 @@ id = "0f2ac74b498b48028cb68387c421e279"
     Ok(())
 }
 
+fn check_for_duplicate_names(manifest: &Manifest) -> Result<(), failure::Error> {
+    let mut names: Vec<String> = Vec::new();
+    let mut duplicate_names: Vec<String> = Vec::new();
+    names.push(manifest.name.to_string());
+    match &manifest.env {
+        Some(environments) => {
+            for (_, environment) in environments.iter() {
+                match &environment.name {
+                    Some(name) => {
+                        if names.contains(name) && !duplicate_names.contains(name) {
```

Comment:
General feedback: this type of repetition detection logic is fantastic to do with Sets: https://doc.rust-lang.org/std/collections/struct.HashSet.html. I would recommend completely replacing `names` and `duplicate_names` with Hashsets instead (it's also more performant do to this, though scale isn't really a problem here).